### ISSUE TITLE
Issues with FOVQC

### ISSUE THREAD

**Issue by SethMagnusJarvis** [Nanostring-Biostats/CosMx-Analysis-Scratch-Space#169]
I'm trying to run FOVQC. I've gone through the vignette up to this point. I've been getting the error:

```
Error in gridmap[, rownames(counts)] : 
  invalid or not-yet-implemented 'Matrix' subsetting

```

I tried making a version where I used a data frame and got what seems like a very similar error, although it took much longer (might just be due to caching or formatting):

`Error in gridmap[, rownames(counts)] %*% counts :    c("not-yet-implemented method for <dgCMatrix> %*% <tbl_df>", "not-yet-implemented method for <dgCMatrix> %*% <tbl>", "not-yet-implemented method for <dgCMatrix> %*% <data.frame>")`

Is this a known issue? I have no idea what's going wrong. I thought it might be due to an update of a package of some sort.

The only difference I can think of is that I had pre-made flat files loaded the files in, then converted them into sparse matrixes in a similar way to how the vignette does.

**Comment by eveilyeverafter** [comment]
This issue might help:
[https://github.com/satijalab/seurat/issues/1044](https://github.com/satijalab/seurat/issues/1044)

**Comment by marazzoli** [comment]
I'm stuck with FOV-QC using our own dataset (based on 4 slides, with up to 8 different samples/pieces of tissue, and up to total ~30ROI per slide).
Here's the error I get entering the code
fovqcresult <- runFOVQC(counts = counts, xy = xy, fov = metadata$FOV, barcodemap = barcodemap, max_prop_loss = 0.3) 
The following FOVs failed QC for one or more barcode positions: NA
Error in (function (cond)  : 
  error in evaluating the argument 'x' in selecting a method for function 'which': subscript out of bounds
Called from: (function (cond) 
.Internal(C_tryCatchHelper(addr, 1L, cond)))(list(message = "subscript out of bounds", 
    call = flags_per_fov_x_reportercycle[f, ], object = c(NA_real_, 
    NA_real_, NA_real_, NA_real_, NA_real_, NA_real_, NA_real_, 
    NA_real_, NA_real_, NA_real_, NA_real_, NA_real_, NA_real_, 
    NA_real_, NA_real_, NA_real_, NA_real_, NA_real_, NA_real_, 
    NA_real_, NA_real_, NA_real_, NA_real_, NA_real_, NA_real_, 
    NA_real_, NA_real_, NA_real_, NA_real_, NA_real_, NA_real_, 
    NA_real_, NA_real_, NA_real_, NA_real_, NA_......

I was able to follow all the previous steps up to now, except I wasn't able to apply the "Custom labeling of tissue" and so I was runnig the intermediate data generated up to them for normalization purposes. 
pls lmk
Tnx

**Comment by eveilyeverafter** [comment]
Could you provide the first few columns and rows of the matrix and the head of the meta, xy, and barcodes? In addition, could you confirm the class each object belongs to?

Is this CosMx SMI 1k or 6k?

**Comment by marazzoli** [comment]
Thank you for your quick reply
heres's the first few rows and columns of the matrix
5 x 10 sparse Matrix of class "dgCMatrix"
  [[ suppressing 10 column names ‘5-HTT’, ‘APOE’, ‘ARG1’ ... ]]
                                                                          
c_4_1_1 25.38 280.54 59.83 195.85 195.39 364.14 1400.14 27.69 106.37 37.85
c_4_1_2 17.82 283.83 57.37 116.46 195.73 143.99 1022.88 15.56  38.88 28.50
c_4_1_3 75.02 360.96 77.47 158.58 205.13 169.81  763.14 26.73  37.53 29.69
c_4_1_4 44.48 216.94 39.05 153.96 206.53 219.18  697.06 23.19  38.49 33.59
c_4_1_5 34.65 338.92 79.03 177.01 263.85 296.01  960.84 34.28  50.02 42.22

head of the meta
     fov  Area AspectRatio CenterX_local_px CenterY_local_px Width Height Mean.GFAP Max.GFAP
   <int> <int>       <num>            <int>            <int> <int>  <int>     <int>    <int>
1:     1  3020        0.65               45               25    79     51         0      182
2:     1  2864        0.51              394               22    89     45         0      126
3:     1  3968        0.93              494               37    81     75        21      314
4:     1  3608        0.97              586               33    65     67        12      781
5:     1  2528        0.51              734               20    81     41         8      388
6:     1  2380        0.64             1040               23    73     47         0        0
   Mean.NeuN Max.NeuN Mean.S6 Max.S6 Mean.Iba1 Max.Iba1 Mean.DAPI Max.DAPI SplitRatioToLocal NucArea
       <int>    <int>   <int>  <int>     <int>    <int>     <int>    <int>             <num>   <int>
1:         0        0       7    398       403     2195       500     1359              0.58       0
2:         0        0       0    225         7      425       536     1929              0.55       0
3:         0        0       2    516         6      245       582     2181              0.76       0
4:         0        0      17    756         1       98       540     1905              0.69       0
5:         9      164     125    878        10      154       307     1007              0.48       0
6:         5      157      32    553        13      276       504     2285              0.45       0
   NucAspectRatio Circularity Eccentricity Perimeter Solidity cell_id assay_type version
            <int>       <num>        <num>     <int>    <num>  <char>     <char>  <char>
1:              0        0.90         0.72       205    14.73 c_4_1_1    protein      v6
2:              0        0.89         0.53       201    14.25 c_4_1_2    protein      v6
3:              0        0.93         0.82       232    17.10 c_4_1_3    protein      v6
4:              0        0.96         0.70       217    16.63 c_4_1_4    protein      v6
5:              0        2.03         0.22       125    20.22 c_4_1_5    protein      v6
6:              0        2.01         0.28       122    19.51 c_4_1_6    protein      v6
   Run_Tissue_name                                  Panel                cellSegmentationSetId
            <char>                                 <char>                               <char>
1:         Slide 1 (1.1) (1.1) Mouse Protein Neuroscience 2e099869-f77a-4234-853e-b23fd3a4bbaf
2:         Slide 1 (1.1) (1.1) Mouse Protein Neuroscience 2e099869-f77a-4234-853e-b23fd3a4bbaf
3:         Slide 1 (1.1) (1.1) Mouse Protein Neuroscience 2e099869-f77a-4234-853e-b23fd3a4bbaf
4:         Slide 1 (1.1) (1.1) Mouse Protein Neuroscience 2e099869-f77a-4234-853e-b23fd3a4bbaf
5:         Slide 1 (1.1) (1.1) Mouse Protein Neuroscience 2e099869-f77a-4234-853e-b23fd3a4bbaf
6:         Slide 1 (1.1) (1.1) Mouse Protein Neuroscience 2e099869-f77a-4234-853e-b23fd3a4bbaf
   cellSegmentationSetName slide_ID CenterX_global_px CenterY_global_px median_RNA RNA_quantile_0.75
                    <char>    <int>             <int>             <int>      <num>             <num>
1:    Initial Segmentation        4             17198            117122   153111.9          379349.7
2:    Initial Segmentation        4             17547            117125   153111.9          379349.7
3:    Initial Segmentation        4             17647            117110   153111.9          379349.7
4:    Initial Segmentation        4             17739            117114   153111.9          379349.7
5:    Initial Segmentation        4             17887            117127   153111.9          379349.7
6:    Initial Segmentation        4             18193            117124   153111.9          379349.7
   RNA_quantile_0.8 RNA_quantile_0.85 RNA_quantile_0.9 RNA_quantile_0.95 RNA_quantile_0.99 nCount_RNA
              <num>             <num>            <num>             <num>             <num>      <num>
1:         574512.8          767932.8         937288.3           1444179           3065237   25557.38
2:         574512.8          767932.8         937288.3           1444179           3065237   21306.35
3:         574512.8          767932.8         937288.3           1444179           3065237   22493.26
4:         574512.8          767932.8         937288.3           1444179           3065237   17813.11
5:         574512.8          767932.8         937288.3           1444179           3065237   21093.90
6:         574512.8          767932.8         937288.3           1444179           3065237   25784.18
   nFeature_RNA median_negprobes negprobes_quantile_0.75 negprobes_quantile_0.8
          <int>            <num>                   <num>                  <num>
1:           67         106564.8                379349.7               574512.8
2:           67         106564.8                379349.7               574512.8
3:           67         106564.8                379349.7               574512.8
4:           67         106564.8                379349.7               574512.8
5:           67         106564.8                379349.7               574512.8
6:           67         106564.8                379349.7               574512.8
   negprobes_quantile_0.85 negprobes_quantile_0.9 negprobes_quantile_0.95 negprobes_quantile_0.99
                     <num>                  <num>                   <num>                   <num>
1:                767932.8               937288.3                 1444179                 3065237
2:                767932.8               937288.3                 1444179                 3065237
3:                767932.8               937288.3                 1444179                 3065237
4:                767932.8               937288.3                 1444179                 3065237
5:                767932.8               937288.3                 1444179                 3065237
6:                767932.8               937288.3                 1444179                 3065237
   nCount_negprobes nFeature_negprobes Area.um2    cell    FOV
              <num>              <int>    <num>  <char> <char>
1:           110.45                  2 43.69187 c_4_1_1   s4f1
2:           105.08                  2 41.43494 c_4_1_2   s4f1
3:           114.35                  2 57.40706 c_4_1_3   s4f1
4:           112.93                  2 52.19876 c_4_1_4   s4f1
5:           150.10                  2 36.57385 c_4_1_5   s4f1
6:           131.40                  2 34.43266 c_4_1_6   s4f1

head of xy
            x_mm     y_mm
c_4_1_1 29.14679 12.33818
c_4_1_2 29.18877 12.33854
c_4_1_3 29.20080 12.33674
c_4_1_4 29.21186 12.33722
c_4_1_5 29.22967 12.33878
c_4_1_6 29.26647 12.33842

head of barcodes
   gene                          barcode
1  AATK ....GG......BB........YY......YY
2  ABL1 ............YYBBYY..........RR..
3  ABL2 ......RR........BB........RR..RR
4 ACACB BB............YY..YY....YY......
5   ACE ....BB....BB........RR..RR......
6 ACKR1 ......RR......GGRR......BB......

It is a CosMX 6k
Pls let me know
Thanks!!!!!

**Comment by fengyun198503** [comment]
I met the following error:
Error in gridmap[, rownames(counts)] %*% counts : 
  non-conformable arguments
Called from: gridmap[, rownames(counts)] %*% counts
Browse[1]> 

Could you help me?
Thanks,

**Comment by patrickjdanaher** [comment]
Hi there,
We can help debug. Please contact Support and work with them to send me the following info:
- the R call that generates the issue
- the data objects you passed to the function
- if non-standard, the full analysis script up to that point.
